### PR TITLE
Support apply_(callable) sugar for CPU NJTs

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128032
* #128031
* __->__ #125416

Example:
```python
nt.apply_(lambda x: x * 2)
```

### CODE REVIEW

File: test/test_nestedtensor.py [hunk 1]
```diff
@@ -4339,6 +4339,28 @@ def forward(self, query, value, offsets):
...
+            return
+
+        before = nt._values.clone().detach()
+
+        nt.apply_(f)
+        expected = f(before)
+        self.assertEqual(expected, nt._values)
+        # apply_ should swap values in-place without appending to autograd graph
+        self.assertIsNone(nt.grad)
+        self.assertIsNone(nt._values.grad)
```
Comment (soulitzer) [inline test/test_nestedtensor.py:4362]: I'm guessing you meant to check .grad_fn :p

**Review by soulitzer — COMMENTED** [review]
Wow didn't even know apply_ existed lol, do you know what its use case is?

**Comment by jbschlosser** [comment]
> Wow didn't even know apply_ existed lol, do you know what its use case is?

Haha, I didn't either until one of the preproc folks pointed it out as a UX gap in NJT. Apparently, they got used to using it for whatever purposes and noticed that it didn't work when moving over to nested tensors.

File: test/test_nestedtensor.py (hunk 1 above)
Comment (jbschlosser) [reply test/test_nestedtensor.py:4362]: right you are :p

**Review by soulitzer — APPROVED** [review]
SGTM

**Pull request closed without merging**